pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/ServerCnx.java

```diff
@@ -192,6 +199,86 @@ protected void handlePartitionMetadataRequest(CommandPartitionedTopicMetadata pa
+        final long consumerId = commandConsumerStats.getConsumerId();
+        
+        if (log.isDebugEnabled()) {
+            log.debug("CommandConsumerStats[requestId = {}, topicName = {}, subscriptionName = {}, consumerId = {}]", requestId, topicName, subscriptionName, consumerId);
+        }
+        
+        Object msg = null;
+        try {
+            PersistentTopic topic;
+            topic = (PersistentTopic) getBrokerService().getTopicReference(topicName);
+            if (topic != null) {
+                if (topic.getSubscriptions().containsKey(subscriptionName)) {
+                    PersistentSubscription subscription = topic.getSubscriptions().get(subscriptionName);
+                    boolean consumerFound = false;
+                    for (Consumer consumer : subscription.getConsumers()) {
+                        if (consumer.consumerId() == consumerId) {
+                            consumerFound = true;
+                            Map<String, String> consumerStatsMap = createConsumerStatsMap(consumer, subscription);
+                            if (log.isDebugEnabled()) {
+                                log.debug("Sending consumer stats [{}] for request id {}", consumerStatsMap, requestId);
+                            }
+                            msg = Commands.newConsumerStatsResponse(consumerStatsMap, requestId, topicName, subscriptionName, consumerId);
+                            break;
+                        }
+                    }
+                    if (!consumerFound) {
+                        log.error("Consumer {} not found for CommandConsumerStats request id {}", consumerId,  requestId);
+                        msg = Commands.newConsumerStatsResponse("Consumer " + consumerId + " not found", requestId, topicName, subscriptionName, consumerId);
+                    }
+                } else {
+                    log.error("Subscription {} not found for CommandConsumerStats request id {}", subscriptionName,  requestId);
+                    msg = Commands.newConsumerStatsResponse("Subscription " + subscriptionName + " not found", requestId, topicName, subscriptionName, consumerId);
+                }
+            } else {
+                log.error("Topic {} not found for CommandConsumerStats request id {}", topicName,  requestId);
+                msg = Commands.newConsumerStatsResponse("Topic " + topicName + " not found",
+                        requestId, topicName, subscriptionName, consumerId);
+            }
+        } catch (Exception e) {
+            log.error("Exception: {}", e);
+            msg = Commands.newConsumerStatsResponse("Exception: " + e, requestId, topicName, subscriptionName, consumerId);
+        } finally {
+            if (msg != null) {
+                ctx.writeAndFlush(msg);
+            }
+        }
+    }
+
+    private Map<String, String> createConsumerStatsMap(Consumer consumer, PersistentSubscription subscription) {
```

Comment:
We need some subscriber level details as well - backlog, subscription type etc 
But I have added a toMap method to convert the Class to a map